atmcfg/atmcfg.go

```diff
@@ -72,9 +77,78 @@ func setDisabledByShardName(out *opsmngr.AutomationConfig, name string, disabled
 	}
 }
 
+func setDisableProcessByProcessNameAndPort(out *opsmngr.AutomationConfig, clusterName string, processesMap map[string]bool, disabled bool) {
+	newDeploymentAuthMechanisms(out)
+	for k, p := range out.Processes {
+		key := fmt.Sprintf("%s:%d", p.Hostname, p.Args26.NET.Port)
+		if _, ok := processesMap[key]; ok && strings.Contains(out.Processes[k].Name, clusterName) {
+			out.Processes[k].Disabled = disabled
+			processesMap[key] = true
+		}
+	}
+}
+
 // Shutdown disables all processes of the given cluster name.
-func Shutdown(out *opsmngr.AutomationConfig, name string) {
-	setDisabledByClusterName(out, name, true)
+func Shutdown(out *opsmngr.AutomationConfig, clusterName string) {
+	setDisabledByClusterName(out, clusterName, true)
+}
+
+// ShutdownClusterAndProcesses disables the entire cluster or its processes. Processes are provided in the format {"hostname:port","hostname2:port2"}.
+func ShutdownClusterAndProcesses(out *opsmngr.AutomationConfig, clusterName string, processes []string) error {
+	if len(processes) == 0 {
+		Shutdown(out, clusterName)
+		return nil
+	}
+	return shutdownProcesses(out, clusterName, processes)
+}
+
+// shutdownProcesses disables processes. Processes are provided in the format {"hostname:port","hostname2:port2"}.
+func shutdownProcesses(out *opsmngr.AutomationConfig, clusterName string, processes []string) error {
+	processesMap := map[string]bool{}
+	for _, hostnameAndPort := range processes {
+		processesMap[hostnameAndPort] = false
+	}
+
+	setDisableProcessByProcessNameAndPort(out, clusterName, processesMap, true)
+
+	return newProcessNotFoundError(clusterName, processesMap)
+}
+
+// newProcessNotFoundError returns an error if a process was not found.
+func newProcessNotFoundError(clusterName string, processesMap map[string]bool) error {
+	err := ""
+	for k, v := range processesMap {
+		if !v {
+			err += fmt.Sprintf("Process %s was not found for Cluster %s.\n", k, clusterName)
```

Comment:
[nit] this is going to be a long repeated message for multiple failures, for a lib this should be simplified to something like `process not found: [list of processes]` and then consumers and improve on it

Reply:
also remember errors on go should not start with caps